.github/workflows/swig-update-autocheck.yml

```diff
@@ -0,0 +1,44 @@
+name: Auto-update SWIG Version
+on:
+  workflow_dispatch:
+  schedule:
+    - cron:  '0 10 * * 1'
+  pull_request_target:
+    branches: [ main ]
+jobs:
+  update-dep:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Check and update SWIG version
+        run: |
+           old_version="$(grep -Po '(?<=SWIG_VERSION )\d+(\.\d+)+' swig_version.cmake)"
```

Comment:
Oops, I think my suggested change before had an extra space and yaml parsers are picky about indentation levels matching.

```suggestion
          old_version="$(grep -Po '(?<=SWIG_VERSION )\d+(\.\d+)+' swig_version.cmake)"
```